### PR TITLE
direwolf: fix build

### LINKED ISSUE #140059: direwolf does not build: libgps API version might be incompatible
### Describe the bug
direwolf does not build.

### Additional context

`/build/source/src/dwgpsd.c:64:2: error: #error libgps API version might be incompatible.`

This seems to have been noticed previously as mentioned in this comment on the upgrade of gpsd from 3.22 -> 3.23:
https://github.com/NixOS/nixpkgs/pull/133216#issuecomment-895943208

### Notify maintainers
@lasandell

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"

 - system: `"x86_64-linux"`
 - host os: `Linux 5.10.52, NixOS, 21.05.3367.fd8a7fd07da (Okapi)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.15`
 - channels(user): `"nixos-21.05pre283367.0a5f5bab0e0, nixpkgs-21.05pre283770.f73da2880a9"`
 - channels(root): `"nixos-21.05.3367.fd8a7fd07da, nixos-hardware, nixos-unstable-21.11pre303315.16105403bdd"`
 - nixpkgs: `/home/anon/.nix-defexpr/channels/nixpkgs`
```

### PR DESCRIPTION
**Pull request by lasandell**
###### Motivation for this change

Fix broken build (resolves #140059) until upstream updates the gpsd version check. I'm overriding the version check but verified that gpsd functionality is still working correctly. Also fixed an issue with espeak and converted small patch to `substituteInPlace` command.

Would be good to backport this since it is currently broken.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/radio/direwolf/default.nix [hunk 1]
```diff
@@ -22,22 +22,23 @@ stdenv.mkDerivation rec {
...
+    substituteInPlace conf/CMakeLists.txt \
+      --replace /etc/udev/rules.d/ $out/lib/udev/rules.d/
     substituteInPlace src/symbols.c \
       --replace /usr/share/direwolf/symbols-new.txt $out/share/direwolf/symbols-new.txt \
       --replace /opt/local/share/direwolf/symbols-new.txt $out/share/direwolf/symbols-new.txt
     substituteInPlace src/decode_aprs.c \
       --replace /usr/share/direwolf/tocalls.txt $out/share/direwolf/tocalls.txt \
       --replace /opt/local/share/direwolf/tocalls.txt $out/share/direwolf/tocalls.txt
     substituteInPlace scripts/dwespeak.sh \
+      --replace /bin/bash /bin/sh \
```
Comment (SuperSandro2000) [inline pkgs/applications/radio/direwolf/default.nix:35]:
```suggestion
      --replace /bin/bash ${bash}/bin/bash \
```

File: pkgs/applications/radio/direwolf/default.nix [hunk 2]
```diff
@@ -22,22 +22,23 @@ stdenv.mkDerivation rec {
     espeak gpsd hamlib perl python3
   ] ++ (optionals stdenv.isLinux [alsa-lib udev]);
 
-  patches = [
-    ./udev-fix.patch
-  ];
-
   postPatch = ''
+    substituteInPlace conf/CMakeLists.txt \
+      --replace /etc/udev/rules.d/ $out/lib/udev/rules.d/
```
Comment (SuperSandro2000) [inline pkgs/applications/radio/direwolf/default.nix:27]: Not sure if this should use placeholder.

Comment (lasandell) [reply pkgs/applications/radio/direwolf/default.nix:27]: What do you mean by placeholder?

File: pkgs/applications/radio/direwolf/default.nix (hunk 1 above)
Comment (lasandell) [reply pkgs/applications/radio/direwolf/default.nix]: I just updated this to use `patchShebangs`, which I forgot was the canonical way to do this.

File: pkgs/applications/radio/direwolf/default.nix (hunk 2 above)
Comment (SuperSandro2000) [reply pkgs/applications/radio/direwolf/default.nix:27]: ``${placeholder "out"}``

Comment (lasandell) [reply pkgs/applications/radio/direwolf/default.nix:27]: I realize this has now been merged, but I looked into it and it seems `placeholder` is only meant to be used in places where output paths are unavailable via environment variables (i.e. anywhere besides phase and pre-/post-phase scripts). Related forum post: https://discourse.nixos.org/t/what-is-the-difference-between-placeholder-out-and-out/4862/2

**Pull request merged**